scripts/end_to_end_test/smart_contract_tests/balance.py

```diff
@@ -101,19 +101,24 @@ def run(options):
 
     api.sync(api.contracts.create(entity1, contract, 10000))
 
+    api.sync(api.tokens.transfer(entity1, contract.address, 1000, 500))
+
     assert contract.query(api, 'query_init_test') == 0
 
-    api.sync(contract.action(api, 'action_test', 10000, [entity1]))
-    assert contract.query(api, 'query_action_test') == 0
+    api.sync(contract.action(api, 'action_test', 200, [entity1]))
+    v = contract.query(api, 'query_action_test')
+    assert 800 <= contract.query(api, 'query_action_test') <= 1000
```

Comment:
Please give all assertions meaningful failure messages. Makes debugging easier.
```suggestion
    v = contract.query(api, 'query_action_test')
    assert 800 <= v <= 1000, \
        'Expected query_action_test result to be between 800 and 1000, found {}'.format(v)
```